src/Core/Framework/Struct/Collection.php

```diff
@@ -234,6 +234,36 @@ public function getIterator(): \Traversable
         yield from $this->elements;
     }
 
+    /**
+     * @param array<mixed> $associativeData
+     *
+     * @return Collection<TElement>
+     */
+    public function addFromAssociative(array $associativeData): self
+    {
+        $expectedClass = $this->getExpectedClass();
+        if ($expectedClass === null) {
+            return $this;
+        }
+
+        $baseObject = (new \ReflectionClass($expectedClass))->newInstanceWithoutConstructor();
+        $hasNecessaryInterface = $baseObject instanceof AssignArrayInterface;
+
+        foreach (\array_filter($associativeData) as $value) {
```

Comment:
Empty values can be important too, we shouldn't make assumptions:
```suggestion
        foreach ($associativeData as $value) {
```

Reply:
Since we are expecting (in fact) objects, `array_filter` will remove empty arrays→ resulting in "empty" objects and maybe `null` values, which will break anyway on `instanceof`

If we want to remove the filter here, we also should remove the empty array check at this point: https://github.com/shopware/shopware/pull/13551/files#diff-7c50a28ddd18be6ca625687b2b54d11963c58d44a75fe0ab45f144daa212beb7R39

In addition, we also have to take the [SDK](https://github.com/shopware/paypal-sdk/blob/trunk/src/Struct/Collection.php#L52) into account. It would be good if we had reasonably consistent behavior across systems.

Reply:
if you only want to check for an empty array, this should be done explicitly.

Reply:
Yes, we should remove the empty array check - back and forth serialization should result in the same structure & it's more likely that the object should be constructed empty as it was serialized to an empty object first.

The SDK is different here, because all its assigns are meant to deserialize PayPal's responses -> we don't own the data, its not the same use-case. Also, when I created the SDK, backwards compatibility was a major factor, nothing besides namespaces should change. `array_filter` was taken over, see [here](https://github.com/shopware/SwagPayPal/blob/6.6.x/src/RestApi/PayPalApiCollection.php#L53).

Reply:
The SDK is not the perfect form of how the deserialization should work, do to many things are kept to keep existing functionally and logic like it was before :sweat_smile: